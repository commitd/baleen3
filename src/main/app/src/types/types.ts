/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { Annot8ComponentDescriptor } from './base-types'
import {
  Annot8ComponentInfo,
  Measurement,
  PipelineTemplate,
  SettingsSchema,
  WithDescription,
  WithName,
  ErrorConfiguration,
} from './server-types'

/**
 * Index of metrics measurements
 */
export interface MeasurementContainer {
  [key: string]: Measurement[]
}

/**
 * Holds all the classes that have metrics associated with them, and their associated measurements
 */
export interface MetricsContainer {
  [key: string]: MeasurementContainer
}

export interface SettingsSchemaExtended extends SettingsSchema {
  /**
   * String version of json uiSchema
   *
   */
  uiSchema: string
}

export interface ComponentInfo extends Annot8ComponentInfo {
  name: string
}

/**
 * Map of server components
 */
export interface ComponentMap {
  [key: string]: ComponentInfo
}

/**
 * Map of settings schema
 */
export interface SettingsMap {
  [key: string]: SettingsSchema
}

/**
 * The full set of customisation details for the server.
 */
export interface ServerDetails {
  /**
   * Index by class name of ComponentInfo that represent the processors available on the server that can be used in pipelines.
   */
  processors: ComponentMap
  /**
   * Index by class name of ComponentInfo that represent the sources available on the server that can be used in pipelines.
   */
  sources: ComponentMap
  /**
   * Index by class name of SettingsSchema that are the settings classes of all sources and processors available on the server that can be used in pipelines.
   */
  settings: SettingsMap
  /**
   * Index by class name of ComponentInfo that represent the orderers available on the server that can be used to order pipelines.
   * <p>
   * NB this is currently generated from a strings to simulate a ComponentInfos
   */
  orderers: ComponentMap
  /**
   * Pipeline templates that can be used to start the process of creating a pipeline
   */
  templates: PipelineTemplate[]
}

export interface ComponentMetadata {
  id: string
  descriptor: Annot8ComponentDescriptor
  info: ComponentInfo | undefined
  schema: SettingsSchema | undefined
  valid: boolean
  configured: boolean
}

/**
 * Mapping of Pipeline descriptor to include extra metadata
 *
 */
export interface PipelineHeaderDescriptor extends WithName, WithDescription {}

/**
 * Mapping of Pipeline descriptor to include extra metadata
 *
 */
export interface PipelineViewDescriptor extends PipelineHeaderDescriptor {
  sourceOrder: string[]
  processorOrder: string[]
  components: Record<string, ComponentMetadata>
  errorConfiguration: ErrorConfiguration
}

/**
 * Utility tracking interface to help maintain the UI key state
 */
export interface PipelineEditDescriptor extends PipelineViewDescriptor {
  nameValid: boolean
  orderer: string
  errors: string[]
}

export type ComponentType = 'Processor' | 'Source'

export const NO_SETTINGS: 'io.annot8.api.settings.NoSettings' =
  'io.annot8.api.settings.NoSettings'

export const NO_OP_ORDERER: 'io.annot8.api.pipelines.NoOpOrderer' =
  'io.annot8.api.pipelines.NoOpOrderer'
