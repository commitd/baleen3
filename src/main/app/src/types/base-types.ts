/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
// We remove the generics for the Settings type from these as we never know what they are
// so do not help with any compile time errors

export interface ComponentDescriptor {
  name: string
  settings?: object
}

export interface Annot8ComponentDescriptor {
  [_class: string]: ComponentDescriptor
}

// Types required to complete generated server types
// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars-experimental
export interface SourceDescriptor extends Annot8ComponentDescriptor {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars-experimental
export interface ProcessorDescriptor extends Annot8ComponentDescriptor {}

/**
 * Interface for describing known metrics, to give better display.
 */
export interface MetricMetadata {
  name?: string
  description?: string
  units?: string
  transform?: (v: number) => number
}

/**
 * Provided stacktrace type to correct optional properties
 */
export interface StackTrace {
  classLoaderName?: string
  className: string
  fileName?: string
  lineNumber: number
  methodName: string
  moduleName?: string
  moduleVersion?: string
  nativeMethod?: boolean
}

/**
 * Provided throwable type to correct optional properties
 */
export interface Throwable {
  cause?: Throwable
  localizedMessage: string
  message: string
  stackTrace: StackTrace[]
  suppressed?: Throwable[]
}
