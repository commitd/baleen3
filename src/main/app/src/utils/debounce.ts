/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
export type Method = (...args: any[]) => void

export function debounce<F extends Method>(
  func: F,
  wait = 200
): (this: ThisParameterType<F>, ...args: Parameters<F>) => void {
  let timeoutId: ReturnType<typeof setTimeout> | undefined

  return function (this: ThisParameterType<F>, ...args: Parameters<F>): void {
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    const context = this

    if (timeoutId !== undefined) {
      clearTimeout(timeoutId)
    }
    timeoutId = setTimeout(() => {
      timeoutId = undefined
      func.apply(context, args)
    }, wait)
  }
}
