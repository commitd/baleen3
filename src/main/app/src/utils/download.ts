/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const downloadJson = (name: string, data: any): void => {
  download(
    name + '.json',
    'data:text/json;charset=utf-8,',
    JSON.stringify(data, null, 2)
  )
}

export const download = (
  name: string,
  mediaType: string,
  data: string
): void => {
  const dataStr = mediaType + encodeURIComponent(data)
  const downloadAnchorNode = document.createElement('a')
  downloadAnchorNode.setAttribute('href', dataStr)
  downloadAnchorNode.setAttribute('download', name)
  document.body.appendChild(downloadAnchorNode) // required for firefox
  downloadAnchorNode.click()
  downloadAnchorNode.remove()
}
