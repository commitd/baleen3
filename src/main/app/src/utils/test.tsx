/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { StylesProvider } from '@material-ui/styles'
import {
  createHistory,
  createMemorySource,
  History,
  LocationProvider,
} from '@reach/router'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { GenerateId } from 'jss'
import React from 'react'
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider'

const generateClassName: GenerateId = (rule, styleSheet) => {
  const prefix =
    styleSheet === undefined || styleSheet.options.classNamePrefix === undefined
      ? ''
      : styleSheet.options.classNamePrefix
  return `${prefix}-${rule.key}`
}

const LightTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="light">{children}</ThemeProvider>
  </StylesProvider>
)

const DarkTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="dark">{children}</ThemeProvider>
  </StylesProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

export function renderWithRouter(
  ui: Readonly<React.ReactElement>,
  route = '/',
  options?: Omit<RenderOptions, 'queries'>,
  delegate = renderLight
): RenderResult & { history: History } {
  const history = createHistory(createMemorySource(route))
  return {
    ...delegate(
      <LocationProvider history={history}>{ui}</LocationProvider>,
      options
    ),
    history,
  }
}
// re-export everything
export * from '@testing-library/react'
// override render method
export { renderLight as render }
export { userEvent }
