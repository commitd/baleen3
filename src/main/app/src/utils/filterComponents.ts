/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { ComponentInfo, ComponentMap } from '../types'
import { parseSearch, standardize } from './text'

export const filterComponents = (
  map: ComponentMap,
  search: string
): Array<ComponentInfo> => {
  const values = Object.values(map)
  if (search.trim() === '') {
    return values
  }

  const searchTerms = parseSearch(search)

  return values.filter(
    (value) =>
      filterBy('artifact', value, searchTerms) ||
      filterBy('description', value, searchTerms) ||
      filterByName(value, searchTerms) ||
      filterByTags(value, searchTerms)
  )
}

function filterBy(
  key: keyof Omit<ComponentInfo, 'tags'>,
  value: ComponentInfo,
  terms: string[]
): boolean {
  // false positive security/detect-object-injection
  // eslint-disable-next-line
  const property = value[key]
  if (property === undefined) {
    return false
  }
  return terms.some((term) => standardize(property).includes(term))
}

function filterByName(value: ComponentInfo, terms: string[]): boolean {
  const property = standardize(value.name)
  return terms.some((term) => property.includes(term))
}

function filterByTags(value: ComponentInfo, terms: string[]): boolean {
  const tags = value.tags?.map((tag) => tag.toLowerCase())
  return terms.some((term) => tags?.includes(term))
}
