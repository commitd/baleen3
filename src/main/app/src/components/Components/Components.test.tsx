/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import React from 'react'
import { Components } from '.'
import { ComponentMap, NO_OP_ORDERER, SettingsMap } from '../../types'
import {
  exampleComponentInfo1,
  exampleSettingSchemaComplex,
  exampleSettingsSchema,
  exampleSourceInfo,
} from '../../types/examples'
import { render, renderDark } from '../../utils/test'

const processors: ComponentMap = {
  [exampleComponentInfo1.componentClass]: exampleComponentInfo1,
}
const sources: ComponentMap = {
  [exampleSourceInfo.componentClass]: exampleSourceInfo,
}
const settings: SettingsMap = {
  [exampleComponentInfo1.settingsClass as string]: exampleSettingsSchema,
  'io.annot8.components.files.sources.FileSystemSourceSettings': exampleSettingSchemaComplex,
}
const orderers = {
  NO_OP_ORDERER: {
    name: 'No Op Orderer',
    componentClass: NO_OP_ORDERER,
    artifact: 'orderer',
  },
}

const serverContext = { processors, sources, orderers, settings }

it('renders without error', () => {
  const { asFragment } = render(<Components {...serverContext} />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without error', () => {
  const { asFragment } = renderDark(<Components {...serverContext} />)
  expect(asFragment()).toMatchSnapshot()
})
