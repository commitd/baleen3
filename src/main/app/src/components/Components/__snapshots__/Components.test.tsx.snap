// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without error 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      style="transform: translateX(-50%);"
    >
      <div
        class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        >
          <div
            class="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="clear"
            class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root MuiIconButton-disabled MuiButtonBase-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    />
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Orderers
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root Styled(WithStyles(ForwardRef(Card)))-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  No Op Orderer
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  io.annot8.api.pipelines.NoOpOrderer
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
            >
              No description
            </p>
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            />
          </div>
        </div>
      </div>
    </div>
     
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Sources
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root Styled(WithStyles(ForwardRef(Card)))-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  File System Source
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  io.annot8.components.files.sources.FileSystemSource
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
            >
              Provides items from the local file system
            </p>
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            />
          </div>
        </div>
      </div>
    </div>
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Processors
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root Styled(WithStyles(ForwardRef(Card)))-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  MGRS
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  io.annot8.components.geo.processors.Mgrs
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
            >
              Extract MGRS coordinates, optionally ignoring MGRS coordinates that could be dates
            </p>
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without error 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      style="transform: translateX(-50%);"
    >
      <div
        class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        >
          <div
            class="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="clear"
            class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root WithNoOpener-root MuiIconButton-disabled MuiButtonBase-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    />
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Orderers
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root Styled(WithStyles(ForwardRef(Card)))-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  No Op Orderer
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  io.annot8.api.pipelines.NoOpOrderer
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
            >
              No description
            </p>
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            />
          </div>
        </div>
      </div>
    </div>
     
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Sources
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root Styled(WithStyles(ForwardRef(Card)))-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  File System Source
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  io.annot8.components.files.sources.FileSystemSource
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
            >
              Provides items from the local file system
            </p>
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            />
          </div>
        </div>
      </div>
    </div>
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Processors
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root Styled(WithStyles(ForwardRef(Card)))-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  MGRS
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  io.annot8.components.geo.processors.Mgrs
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root WrappedTypography-root WrappedTypography-root MuiTypography-body1"
            >
              Extract MGRS coordinates, optionally ignoring MGRS coordinates that could be dates
            </p>
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
